sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -125,9 +125,15 @@
     Mono<Response<AreaCodes>> getAllAreaCodesWithResponse(
         String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions,
         Context context) {
-        Objects.requireNonNull(locationType, "'locationType' cannot be null.");
-        Objects.requireNonNull(countryCode, "'countryCode' cannot be null.");
-        Objects.requireNonNull(phonePlanId, "'phonePlanId' cannot be null.");
+        if (locationOptions == null) {
+            Mono.error(new NullPointerException("'locationType' cannot be null."));
+        }
+        if (countryCode == null) {
+            Mono.error(new NullPointerException("'countryCode' cannot be null."));
+        }
+        if (phonePlanId == null) {
+            Mono.error(new NullPointerException("'phonePlanId' cannot be null."));
+        }
```

Comment:
Like you did in the other method, move these inside the try block instead.. as a result the exception will be caught and return as monoError in the exception.. this needs to be done for all the methods in this client

Reply:
Just surrond all the methods where there could be an exeception into a try